go/grpcwebproxy/main.go

```diff
@@ -49,39 +51,44 @@ func main() {
 	// gRPC-Web compatibility layer with CORS configured to accept on every
 	wrappedGrpc := grpcweb.WrapServer(grpcServer, grpcweb.WithCorsForRegisteredEndpointsOnly(false))
 
-	// Debug server.
-	debugServer := http.Server{
-		WriteTimeout: *flagHttpMaxWriteTimeout,
-		ReadTimeout:  *flagHttpMaxReadTimeout,
-		Handler: http.HandlerFunc(func(resp http.ResponseWriter, req *http.Request) {
-			wrappedGrpc.ServeHTTP(resp, req)
-		}),
-	}
-	http.Handle("/metrics", promhttp.Handler())
-	debugListener := buildListenerOrFail("http", *flagHttpPort)
-	go func() {
-		logrus.Infof("listening for http on: %v", debugListener.Addr().String())
-		if err := debugServer.Serve(debugListener); err != nil {
-			errChan <- fmt.Errorf("http_debug server error: %v", err)
+	if *runHttpServer {
+		// Debug server.
+		debugServer := http.Server{
```

Comment:
Could we please pull this into a separate function to reduce the duplication between the `http` and `tls` version? Something like `func buildDebugServer(...) http.Server {...}`